tests/cpp/test_multidevice_transformer.cpp

```diff
@@ -431,21 +444,63 @@ LinearBackwardsResult sharded_linear_backwards(
   return {grad_x, grad_w, grad_b};
 }
 
+// Forward layer_norm with cached mean_bcast and invstd tensors to avoid
+// recomputing Welford. For use in backwards pass.
+TensorView* layer_norm_with_cached_statistics(
+    TensorView* x,
+    TensorView* mean_bcast,
+    TensorView* invstd,
+    const std::vector<int64_t>& norm_shape,
+    TensorView* weight,
+    TensorView* bias) {
+  const int64_t kNumberOfDims =
+      (int64_t)TensorDomain::noReductions(x->getLogicalDomain()).size();
+  const int64_t kOuterNumDims = kNumberOfDims - norm_shape.size();
+  std::vector<bool> outer_broadcast_mask(kNumberOfDims, false);
+  for (const auto idx : c10::irange(kOuterNumDims)) {
+    outer_broadcast_mask[idx] = true;
+  }
+
+  auto x_sub_mean = sub(x, mean_bcast);
+  auto y = mul(x_sub_mean, invstd);
+
+  // Optional: norm * weight
+  if (weight != nullptr) {
```

Comment:
AFAICS, all callers pass in a non-null weight and bias, so this check can be removed for simplicity.